### PR TITLE
Initial check-in of spandex log scraper

### PR DESCRIPTION
**Pull request by rlvoyer**
This includes a new subproject (in Python) for periodically extracting Spandex logs from Sumo and dumping them to S3 for analysis.

### CODE REVIEW

File: spandex-log-scraper/scrape_logs.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    one_week_ago = now - timedelta(weeks=1)
+    return one_week_ago.replace(hour=0, minute=0, second=0, microsecond=0)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Extract JSON request logs from the Sumo Logic API")
+
+    parser.add_argument(
+        "-f", "--from_date", type=isodate, default=one_week_ago(),
```
Comment (kavedder) [inline spandex-log-scraper/scrape_logs.py]: Total nit, but I think argparse automatically accepts single-letter single-dash args if they're unambiguous.

File: spandex-log-scraper/scrape_logs.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    logging.basicConfig(format='%(message)s', level=logging.INFO)
+
+    args = parse_args()
+
+    from_date = args.from_date
+    to_date = args.to_date
+
+    s3_client = S3Client.connect_with_profile(args.aws_profile)
+    s3_bucket = args.s3_bucket
+    s3_key = "{}_{}.logs.json".format(from_date.isoformat(), to_date.isoformat())
```
Comment (kavedder) [inline spandex-log-scraper/scrape_logs.py]: `{}_{}` made me giggle.

File: spandex-log-scraper/scrape_logs.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    parser.add_argument("--sumo_access_key", help="The access key to use for the Sumo Logic API")
+
+    return parser.parse_args()
+
+
+REQUEST_PATH_RE = re.compile(
+    r"/suggest/(?P<dataset_id>alpha\.[0-9]+)/(?P<pub_stage>[A-Za-z]+)/(?P<column_id>[0-9a-z]{4}-[0-9a-z]{4})")
+
+
+def extract(message, dataset_id_fxf_map, fxf_domain_map):
```
Comment (kavedder) [inline spandex-log-scraper/scrape_logs.py]: Wait...it doesn't look like this function ever gets called?

Comment (rlvoyer) [inline spandex-log-scraper/scrape_logs.py]: Well, hot diggity. Good catch! Copy/paste error. Will fix.

File: spandex-log-scraper/scrape_logs.py (hunk 2 above)
Comment (rlvoyer) [inline spandex-log-scraper/scrape_logs.py]: 😄

**Pull request merged**